File: SDKs/googleapis.com/androidpublisher/v3/java/lib/src/main/java/openapisdk/models/shared/ManagedProductTaxAndComplianceSettings.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ManagedProductTaxAndComplianceSettings
 * Details about taxation and legal compliance for managed products.
**/
public class ManagedProductTaxAndComplianceSettings {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("eeaWithdrawalRightType")
    public ManagedProductTaxAndComplianceSettingsEeaWithdrawalRightTypeEnum eeaWithdrawalRightType;
    public ManagedProductTaxAndComplianceSettings withEeaWithdrawalRightType(ManagedProductTaxAndComplianceSettingsEeaWithdrawalRightTypeEnum eeaWithdrawalRightType) {
        this.eeaWithdrawalRightType = eeaWithdrawalRightType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("taxRateInfoByRegionCode")
    public java.util.Map<String, RegionalTaxRateInfo> taxRateInfoByRegionCode;
    public ManagedProductTaxAndComplianceSettings withTaxRateInfoByRegionCode(java.util.Map<String, RegionalTaxRateInfo> taxRateInfoByRegionCode) {
        this.taxRateInfoByRegionCode = taxRateInfoByRegionCode;
        return this;
    }
}

File: SDKs/googleapis.com/androidpublisher/v3/python/src/sdk/models/shared/managedproducttaxandcompliancesettings.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import regionaltaxrateinfo as shared_regionaltaxrateinfo

class ManagedProductTaxAndComplianceSettingsEeaWithdrawalRightTypeEnum(str, Enum):
    WITHDRAWAL_RIGHT_TYPE_UNSPECIFIED = "WITHDRAWAL_RIGHT_TYPE_UNSPECIFIED"
    WITHDRAWAL_RIGHT_DIGITAL_CONTENT = "WITHDRAWAL_RIGHT_DIGITAL_CONTENT"
    WITHDRAWAL_RIGHT_SERVICE = "WITHDRAWAL_RIGHT_SERVICE"


@dataclass_json
@dataclasses.dataclass
class ManagedProductTaxAndComplianceSettings:
    r"""ManagedProductTaxAndComplianceSettings
    Details about taxation and legal compliance for managed products.
    """
    
    eea_withdrawal_right_type: Optional[ManagedProductTaxAndComplianceSettingsEeaWithdrawalRightTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('eeaWithdrawalRightType') }})
    tax_rate_info_by_region_code: Optional[dict[str, shared_regionaltaxrateinfo.RegionalTaxRateInfo]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('taxRateInfoByRegionCode') }})
    
